app/src/main/java/com/example/byitscover/scrapers/GoodreadsScraper.java

```diff
@@ -44,15 +46,39 @@ public GoodreadsScraper() {
      * @throws IOException if the text is not found to be returned from the site
      */
     public List<BookListing> scrape(Query query) throws IOException {
-        //Get google url, make sure there are no newlines, and connect to it
-        List<Result> results = ScraperHelper.googleAPISearch(ScraperConstants.GOODREADS, query);
-        String searchingUrl = ScraperHelper.getTopResultUrl(results);
-        searchingUrl.replaceAll("[\\n]", "");
-        Document bookDocument = Jsoup.connect(searchingUrl).get();
-        System.out.println(searchingUrl);
+        String url = getUrlWithQuery(query);
+        System.out.println(url);
+        Document document = Jsoup.connect(url).get();
+        Elements bookLinks = document.getElementsByClass("bookTitle");
 
+        BookListing firstListing = getListingFromElement(bookLinks, 0, query);
```

Comment:
Check if `bookLinks` is empty

Reply:
The nice and tidy loop you had should take care of that